src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -140,22 +149,39 @@ public void AssignRowsAndInitializeDisplay()
             foreach (ObservedProvider provider in _providers.Values.OrderBy(p => p.KnownProvider == null).ThenBy(p => p.Name)) // Known providers first.
             {
                 Console.WriteLine($"[{provider.Name}]"); row++;
+                int rowCount = 0;
+
                 foreach (ObservedCounter counter in provider.Counters.Values.OrderBy(c => c.DisplayName))
                 {
+                    if(rowCount >= _consoleHeight - 5)
+                    {
+                        break;
+                    } 
+
+                    _maxNameLength = Math.Max(Math.Min(80, _consoleWidth) - (CounterValueLength + Indent + 1), 0);
```

Comment:
I would add a comment saying that our goal is to truncate the name so that we don't line wrap as long as the console width is >= CounterValueLength + Indent + 1 characters. That will probably help the reader make more sense of this calculation.

Reply:
Added